internal/api/handlers/orders.go

```diff
@@ -0,0 +1,210 @@
+	acc := order.Bonus.Accrual.InexactFloat64()
+	time := order.Uploaded.Format(time.RFC3339)
+	return &OrderResponse{Onumber: order.Onumber, Status: order.Status.String(), Accrual: acc, Uploaded: time}
+}
+
+type HandlerOrder struct {
+	market   *services.Market
+	conf     *config.Config
+	observer *services.Observer
+}
+
+func NewHandlerOrder(conf *config.Config, market *services.Market, observer *services.Observer) *HandlerOrder {
+
+	return &HandlerOrder{market: market, conf: conf, observer: observer}
+}
+
+func (u *HandlerOrder) SetOrder(res http.ResponseWriter, req *http.Request) {
+
+	//get UserID from cxt values
+	ctxConfig := req.Context().Value(config.CtxConfig{}).(config.CtxConfig)
+
+	// Check from middleware is user authorized
+	if !ctxConfig.IsRegistered() {
+		http.Error(res, "JWT not found.", http.StatusUnauthorized)
+		return
+	}
+
+	userID := ctxConfig.GetUserID()
+
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		// 400
+		http.Error(res, "Cat't read body data", http.StatusBadRequest)
+		return
+	}
+	defer req.Body.Close()
+
+	onumber := string(body)
+	zap.S().Infoln("Set Order for user: ", userID, " Order: ", onumber)
+	// Create order
+	order := model.NewOrder(userID, onumber, false, &decimal.Zero, &decimal.Zero)
+
+	isValid := order.IsValid()
+	if !isValid {
+		// 422
+		errt := "Order nuber not vaild."
+		zap.S().Debugln(errt, onumber)
+		http.Error(res, errt, http.StatusUnprocessableEntity)
+		return
+	}
+
+	isExist, err := u.market.SetOrder(req.Context(), false, order)
+	if !isExist && err != nil {
+		// 500
+		errt := "Get error during save new order."
+		zap.S().Error(errt, onumber)
+		http.Error(res, errt, http.StatusInternalServerError)
+		return
+	}
+
+	if isExist {
+
+		//Check if order created befower with withdraw as prepaid
+		isPreorder, err := u.market.IsPreOrder(req.Context(), userID, onumber)
+		if err != nil {
+			errt := "Get error during preorder search."
+			zap.S().Error(errt, err, onumber)
+			http.Error(res, errt, http.StatusInternalServerError)
+			return
+		}
+
+		if isPreorder {
+			// Move prepaid preOreder to regular order.
+			err = u.market.MovePreOrder(req.Context(), order)
+			if err != nil {
+				errt := "Get error during preorder update."
+				zap.S().Debugln(errt, onumber)
+				http.Error(res, errt, http.StatusInternalServerError)
+				return
+			}
+		}
+
+		// Is Existed for this user.
+		isExistUser, err := u.market.IsExistForUser(req.Context(), userID, onumber)
+		if err != nil {
+			errt := "Get error during search duplicated order for user."
+			zap.S().Error(errt, onumber)
+			http.Error(res, errt, http.StatusInternalServerError)
+			return
+		}
+		if isExistUser {
+			// 200 alredy created for user
+			errt := "Order duplicated for User."
+
+			zap.S().Debugln(errt, onumber)
+			http.Error(res, errt, http.StatusOK)
+			return
+		}
+
+		// Is Existed for others user.
+		isExistOther, err := u.market.IsExistForOtherUsers(req.Context(), userID, onumber)
```

Comment:
```suggestion
		isExist, err := u.market.OrderExist(req.Context(), onumber)
```
В данном случае уже не важно кем был создан заказ. Поэтому фильтр лишний

Reply:
Да, убрал, это лишнее.